crates/tools/gnullvm/src/main.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::BTreeMap;
+use std::io::prelude::*;
+
+fn main() {
+    let mut cmd = std::process::Command::new("where");
+    cmd.arg("llvm-dlltool.exe");
+
+    let output = cmd.output().unwrap();
+
+    if !output.status.success() {
+        println!("llvm-dlltool.exe not found");
+        return;
+    }
+
+    let target = std::env::args().collect::<Vec<_>>();
+    let mut platform_and_target = vec![];
+    if target.iter().any(|x| x == "x86_64") || target.iter().any(|x| x == "all") {
+        platform_and_target.push(("x86_64_gnullvm", "i386:x86-64"));
+    }
+    if target.iter().any(|x| x == "aarch64") || target.iter().any(|x| x == "all") {
+        platform_and_target.push(("aarch64_gnullvm", "arm64"));
+    }
+    if platform_and_target.is_empty() {
+        println!("Please specify at least one architecture or use 'all' argument");
+        return;
+    };
+
+    let files = vec![metadata::reader::File::new("crates/libs/metadata/default/Windows.winmd").unwrap(), metadata::reader::File::new("crates/libs/metadata/default/Windows.Win32.winmd").unwrap(), metadata::reader::File::new("crates/libs/metadata/default/Windows.Win32.Interop.winmd").unwrap()];
+    let reader = &metadata::reader::Reader::new(&files);
+
+    let mut libraries = BTreeMap::<String, BTreeMap<String, usize>>::new();
+    let root = reader.tree("Windows.Win32", &[]).expect("`Windows` namespace not found");
+    for tree in root.flatten() {
+        if let Some(apis) = reader.get(metadata::reader::TypeName::new(tree.namespace, "Apis")).next() {
+            for method in reader.type_def_methods(apis) {
+                let impl_map = reader.method_def_impl_map(method).expect("ImplMap not found");
+                let scope = reader.impl_map_scope(impl_map);
+                let library = reader.module_ref_name(scope).to_lowercase();
+                let params = reader.method_def_size(method);
+                libraries.entry(library).or_default().insert(reader.method_def_name(method).to_string(), params);
+            }
+        }
+    }
```

Comment:
All of this can now be replaced with `libraries()` from the `tool_lib` crate.

Reply:
Copied this code from `tool_gnu`, should I also update it as well (in another PR)?

Reply:
@mati865 Is your choice 👍